plugins/header_rewrite/header_rewrite.cc

```diff
@@ -42,8 +42,27 @@ namespace header_rewrite_ns
 DbgCtl dbg_ctl{PLUGIN_NAME_DBG};
 DbgCtl pi_dbg_ctl{PLUGIN_NAME};
 
-std::once_flag initHRWLibs;
-PluginFactory  plugin_factory;
+std::once_flag   initHRWLibs;
+PluginFactory    plugin_factory;
+std::atomic<int> defer{0};
```

Comment:
Unfortunately, I don't think it can work to have a global defer counter.  A per-transaction defer counter is needed.